src/vsr/replica.zig

```diff
@@ -114,12 +134,41 @@ pub fn Replica(
         view_normal: u32,
 
         /// The current status, either normal, view_change, or recovering:
-        /// TODO Don't default to normal, set the starting status according to the journal's health.
-        status: Status = .normal,
+        status: Status = .recovering,
 
         /// The op number assigned to the most recently prepared operation:
         op: u64,
 
+        /// The op of the highest checkpointed message.
+        // TODO Update this to use LSM storage.
+        // TODO Refuse to store/ack any op>op_checkpoint+journal_slot_count.
+        // TODO enforce invariant op≥op_checkpoint
+        op_checkpoint: u64 = 0,
+
+        /// After recovering the WAL, there are 2 possible outcomes:
+        ///
+        /// * All entries valid. The highest op is certain, and safe to set as `replica.op`.
+        /// * One or more entries are faulty. The highest op isn't certain — it may be one of the
+        ///   broken entries.
+        ///
+        /// The replica must refrain from repairing any faulty slots until the highest op is known.
+        /// Otherwise, if we were to repair a slot while uncertain of `replica.op`:
+        ///
+        /// * we may nack an op that we shouldn't, or
+        /// * we may replace a prepared op that we were guaranteeing for the primary, potentially
+        ///   forking the log.
+        ///
+        /// The replica may learn its op (set `op_known` to true) by any of the following
+        /// (in approximate descending order of likelihood):
+        ///
+        /// * recovering an intact journal (faulty.count=0)
+        /// * receiving a `prepare` that advances the op
+        /// * participating in a view change
+        /// * the `recovery_response`'s headers repaired all faulty slots
+        /// * receiving a `commit` with an `op` we have
+        /// * receiving a header/prepare that repairs the last faulty entry
```

Comment:
```suggestion
        /// The replica may have confidence in its highest op by any of the following means:
        ///
        /// * recovering its journal intact (faulty.count=0),
        /// * recovering its journal with all faulty bits only between `op_checkpoint` and `op`, or by
        /// * completing the recovery protocol (which it must also do to learn its `view`).
```

When we switch to using the SuperBlock to record the replica's latest view number, we can reduce our recovery protocol to only requesting a `start_view` message from the primary of the current view, and then exit the `recovering` status as soon as this is received.

Reply:
Resolving; `op_known` is removed.